client/login.go

```diff
@@ -8,17 +8,38 @@ import (
 	"github.com/moby/moby/api/types/registry"
 )
 
+type RegistryLoginOptions struct {
+	Username      string
+	Password      string
+	ServerAddress string
+	IdentityToken string
+	RegistryToken string
+}
+
+// RegistryLoginResult holds the result of a RegistryLogin query.
+type RegistryLoginResult struct {
+	Auth registry.AuthenticateOKBody
```

Comment:
I'll open a follow-up PR to rename this type to something cleaner (`AuthenticateResponse` probably?)